woocommerce/woocommerce
# [admin-layout] Add Gutenberg sidebar elements

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52520

This PR moves elements from Product Editor (gathered from the WP's Site Editor originally) to `@woocommerce/admin-layout` so the Settings Editor could make use of the same elements.

* Layout
* Sidebar
* SidebarNavigationItem
* SidebarNavigationScreen
* SiteHub
* get-gutenberg-version.ts
* lock-unlock.ts



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. on a JN site, add the Gutenberg plugin and activate
2. Make sure Gutenberg experiments are turned on, Gutenberg > Experiments > Custom dataviews
3. Disable Gutenberg so we can test this case.
4. Turn on the Product Editor WooCommerce > Settings > Advanced > Feature > Try the new Product Editor
5. Turn on the feature flags `settings` and `product-data-views`, Tools > WCA Test Helper > Features
6. Visit the new Product Editor, Products > All Products ( new ). You should see a message saying Gutenberg needs to be enabled.
7. Visit the new Settings, WooCommerce > Settings. You should see a similar message.
8. Since we are unlocking the same packages from WP used in the Customizer, we need to make sure the Customizer still functions, WooCommerce > Home > Task List > Customize your store
9. Enable Gutenberg, see the new Settings screen load with a Sidebar and placeholder content
10. Go back to the new Product Editor***. Click on the eye icon in the top left
<img width="214" alt="image" src="https://github.com/user-attachments/assets/f0daa466-b635-4658-a8a0-3e7553535de8">

11. See the sidebar open up
12. Finally check the Customizer again. It should work as expected.
13. Lastly check the new file structure. Does it all make sense? Does the naming work?

*** There's currently an error on trunk for the Product Editor on JN, but this step works locally. See https://github.com/woocommerce/woocommerce/issues/52635
<img width="625" alt="image" src="https://github.com/user-attachments/assets/508bf891-f646-423c-af81-d90a322cd1e8">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Move Site Editor components to `@woocommerce/admin-layout` package

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
